### ISSUE TITLE
Bad Request 400 on workspace accounts

### ISSUE THREAD

**Issue by akalderon** [sabeechen/hassio-google-drive-backup#1111]
Since updating HA to 18.1 / Supervisor 11.4, Google Backup is encountering this error consistently both when attempting a scheduled backup or even if I manually try to action a 'sync'.  

I re-authorized with Google successfully but the error continue to repeat itself.  Local backups are fine.  Please advise?

12-10 15:31:23 ERROR [backup.model.coordinator]    
  addon/backup/model/coordinator.py:198 (_sync)   
  addon/backup/model/model.py:250 (sync)   
  addon/backup/drive/drivesource.py:218 (save)   
  addon/backup/drive/drivesource.py:205 (save)   
  addon/backup/drive/driverequests.py:352 (create)   
  addon/backup/drive/driverequests.py:316 (create)   
  addon/backup/drive/driverequests.py:388 (retryRequest)   
  addon/backup/creds/driverequester.py:51 (request)   
  /site-packages/aiohttp/client_reqrep.py:1059 (raise_for_status)   
aiohttp.client_exceptions.ClientResponseError: 400, message='Bad Request', url=URL('https://www.googleapis.com/upload/drive/v3/files/?uploadType=resumable&supportsAllDrives=true&upload_id=<*** Removed in this post out of security concerns ***')   

12-10 15:31:23 INFO [backup.model.coordinator] I'll try again in 2 hours

**Comment by rickpaniagua** [comment]
Same error, but nothing changed as far HA updates, nor Google Drive Backup versions. Everything is the same, but uploads are now failing, as of last night or yesterday some time.

I can manual and successfully upload files using the google drive web UI (drag drop and using the upload button). 

I also tested uploading a file from an iPhone, works fine.  

I'm using the same account that HA is using. 

It's just not working from this add-on, nor from my android device with the native Google Drive app.  Same account, same target folder.  

Could the /upload/drive/v3/files/ API be down on Google's end?  

This is using a Google Workspace account if that matters.   

I switched accounts to my gmail and the manual file upload works!  Are you using a Workspace account? 

```
12-10 20:21:54 ERROR 
  addon/backup/model/coordinator.py:198 (_sync)
  addon/backup/model/model.py:250 (sync)
  addon/backup/drive/drivesource.py:218 (save)
  addon/backup/drive/drivesource.py:205 (save)
  addon/backup/drive/driverequests.py:352 (create)
  addon/backup/drive/driverequests.py:316 (create)
  addon/backup/drive/driverequests.py:388 (retryRequest)
  addon/backup/creds/driverequester.py:51 (request)
  /site-packages/aiohttp/client_reqrep.py:1059 (raise_for_status)
aiohttp.client_exceptions.ClientResponseError: 400, message='Bad Request', url=URL('https://www.googleapis.com/upload/drive/v3/files/?uploadType=resumable&supportsAllDrives=true&upload_id=<truncated>')
12-10 20:21:54 INFO I'll try again in 2 hours
12-10 22:21:54 INFO Syncing Backups
12-10 22:21:54 INFO A backup directory file was modified, we'll check for new backups soon.
```

Core 2024.10.1
Supervisor 2024.11.4
Operating System 13.1
Frontend 20241002.2

**Comment by akalderon** [comment]
Yes. I am on a Google Workspace account as well.  Backing up to a Shared Drive.  This has never been a problem until 3 days ago. Nothing changed other than upgrading to HA 18.1 and now to 18.2.  Still same error experienced.

**Comment by rickpaniagua** [comment]
Same: Workspace account, backing up to a shared drive.  

I generated a temporary access token and uploaded the same backup file that's failing (among others that are stacking up now) using the Google Drive v3 API in python.  I tested with and without the `supportsAllDrives=True` option which allows uploading to shared drives (and which we can see this add-on uses) and the backup tar file uploaded fine.  

I tried reauthorizing as well, which didn't help.

I am nowhere near my storage quota.

**Comment by sabeechen** [comment]
Sorry for the delayed response, I'm triaging my backlog of issues and trying to determine what needs addressing.

I just tried using the addon with a low-privledge google workspace account and couldn't reproduce this.  All the reports for this error happened on dec 12th and I'm hoping it was just a blip on Google's end.

Could someone confirm if this is still happening?

**Comment by rickpaniagua** [comment]
Meant to report back here that it started working a few days later.  I found that Google wasn't accepting the default chunk size.  It needed to be increased.  I was updating test cases to prepare a PR to increase the the default size when all of a sudden it began working again. So yes, I believe it was an issue on Google's side that they quietly corrected.

**Comment by sabeechen** [comment]
Heh, that sounds like them
